### PR TITLE
Various minor improvements

### PR DESCRIPTION
**Pull request by tortmayr**
#### What it does

- Merge *-default.ts implementation files of `FeedbackActionDispatcher` and `HelperLineManager` with the the interface definition files. The split is no longer necessary since we resolved potential circular dependency issues
- Ensure that we consistently use the `LazyInjector` over dedicated async providers and deprecate existing provider injection

#### Changelog

-   [ ] This PR should be mentioned in the changelog
-   [ ] This PR introduces a breaking change (if yes, provide more details below for the changelog and the migration guide)

### CODE REVIEW

File: packages/client/src/base/focus/focus-tracker.ts [hunk 1]
```diff
@@ -29,7 +30,7 @@ export interface FocusChange {
  * Allows querying of the current focus state and the focused root diagram element and the currently focused element within the diagram.
  */
 @injectable()
-export class FocusTracker implements IActionHandler {
+export class FocusTracker implements IActionHandler, IDiagramStartup {
     protected inActiveCssClass = 'inactive';
     protected _hasFocus = true;
```
Comment (martin-fleck-at) [inline packages/client/src/base/focus/focus-tracker.ts:35]: I wonder if that should really be true now that we actually track the focus change in the beginning?

File: packages/client/src/base/focus/focus-tracker.ts [hunk 2]
```diff
@@ -57,12 +58,20 @@ export class FocusTracker implements IActionHandler {
...
+    postModelInitialization(): MaybePromise<void> {
+        this.handleFocusChange();
+    }
+
     handle(action: Action): void | Action | ICommand {
         if (!FocusStateChangedAction.is(action)) {
             return;
         }
 
         this._hasFocus = action.hasFocus;
```
Comment (martin-fleck-at) [inline packages/client/src/base/focus/focus-tracker.ts:70]: This should probably be initialized correctly somehow.

**Review by martin-fleck-at — CHANGES_REQUESTED** [review]
Change looks good to me overall but I wonder whether the focus tracking needs to be correctly initialized. Also the PR has merge conflicts. Other than that, I like the cleanup and various improvements, thank you Tobias!

**Comment by tortmayr** [comment]
Thanks for the review Martin. I decided to revert the focus tracker fix for now.
As you said, simply setting the focus to true on init is probably not the right approach and we need to come up with a proper solution. Since this potentially also will affect all integration projects i decided to handle this in a follow-up.
It`s a non-critical release anyways so we can fix this issue later in one of the next release cycles.

**Review by martin-fleck-at — APPROVED** [review]
I'm fine with handling the focus tracking in a follow-up issue. Other than that, everything looks good to me, thank you Tobias!

**Pull request merged**